### DISCUSSION TITLE
nats client request msg - ignores replyto subject

### DISCUSSION THREAD

**Discussion by blinkinglight** [nats-io/nats.go#1597] [Ideas]
https://github.com/nats-io/nats.go/blob/main/nats.go#L3986
https://github.com/nats-io/nats.go/blob/main/nats.go#L4011
https://github.com/nats-io/nats.go/blob/main/nats.go#L4063
https://github.com/nats-io/nats.go/blob/main/nats.go#L3952

what i expect: if i pass nats.Msg - use all of it not some parts.

**Comment by ripienaar** [comment]
Request function mean specifically that we handle the replies and process the reply message, those functions can’t use the supplied reply subject. 

You should use PublishMsg() to send it un modified.

**Reply by blinkinglight** [reply]
thats kind of "problem" that i need to copy/paste logic from "https://github.com/nats-io/nats.go/blob/main/nats.go#L4063" instead of set for msg.replyto = my.specific.inbox 
because publishmsg does that? but requestmsg doesnt?

**Reply by ripienaar** [reply]
Inboxes should really be unique per request unless you do something special - in which case our logic wouldnt really work.

So what you should consider is setting a custom inbox prefix so it'll get `my.specific.inbox.>`

**Reply by blinkinglight** [reply]
i had trie to pass inbox prefix to https://github.com/nats-io/nats.go/blob/main/nats.go#L3986 but it will be override by https://github.com/nats-io/nats.go/blob/main/nats.go#L4063 through this https://github.com/nats-io/nats.go/blob/main/nats.go#L4020

**Reply by ripienaar** [reply]
You set https://github.com/nats-io/nats.go/blob/33316cdf884166bf6231192fa88304bb6533367c/nats.go#L1359 when creating the `nc` and all request-reply will use that prefix for the life of the nc
